### PR TITLE
Events for signing and verifying signatures

### PR DESCRIPTION
**Pull request by matejpavlovic**
Signed-off-by: Matej Pavlovic <matopavlovic@gmail.com>

### CODE REVIEW

File: workers.go [hunk 1]
```diff
@@ -298,6 +298,28 @@ func (n *Node) processCryptoEvents(eventsIn *events.EventList) (*events.EventLis
...
+			// Convenience variables
+			ve := e.VerifyNodeSig
+			nodeID := t.NodeID(ve.NodeId)
+
+			err := n.modules.Crypto.VerifyNodeSig(ve.Data, ve.Signature, t.NodeID(ve.NodeId))
+			// Create result event, depending on verification outcome.
+			if err == nil {
+				eventsOut.PushBack(events.NodeSigVerified(true, "", nodeID, ve.Origin))
+			} else {
+				eventsOut.PushBack(events.NodeSigVerified(true, err.Error(), nodeID, ve.Origin))
```
Comment (sergefdrv) [inline workers.go:321]:
```suggestion
				eventsOut.PushBack(events.NodeSigVerified(false, err.Error(), nodeID, ve.Origin))
```

File: protos/isspb/isspb.proto [hunk 2]
```diff
@@ -43,9 +43,11 @@ message Checkpoint {
...
 message SBInstanceMessage {
   oneof type {
-    isspbftpb.Preprepare pbft_preprepare = 1;
-    isspbftpb.Prepare    pbft_prepare    = 2;
-    isspbftpb.Commit     pbft_commit     = 3;
+    isspbftpb.Preprepare        pbft_preprepare         = 1;
+    isspbftpb.Prepare           pbft_prepare            = 2;
+    isspbftpb.Commit            pbft_commit             = 3;
+    isspbftpb.SignedViewChange  pbft_signed_view_change = 4;
+    isspbftpb.PreprepareRequest pbft_preprepare_request = 5;
```
Comment (sergefdrv) [inline protos/isspb/isspb.proto:50]: Looks unrelated to this PR...

File: protos/isspbftpb/isspbftpb.proto [hunk 3]
```diff
@@ -36,34 +36,39 @@ message Commit {
...
 
-message PreprepareHashOrigin {
-  Preprepare preprepare = 1;
+// ============================================================
+// Events
+// ============================================================
+
+message ReqWaitReference {
+  uint64   sn = 1;
+  uint64 view = 2;
```
Comment (sergefdrv) [inline protos/isspbftpb/isspbftpb.proto:71]: What is this for?

**Review by sergefdrv — CHANGES_REQUESTED** [review]

File: protos/isspbftpb/isspbftpb.proto (hunk 3 above)
Comment (matejpavlovic) [reply protos/isspbftpb/isspbftpb.proto:71]: It's part of the PBFT code that uses signatures for view changes. It's meant to make sure that requests are properly preprepared across views. It is not directly linked to signatures themselves, but the PBFT algorithm (after introducing signatures and view changes) needs it.

File: protos/isspb/isspb.proto (hunk 2 above)
Comment (matejpavlovic) [reply protos/isspb/isspb.proto:50]: It's a message required by the view change implementation, for the new primary to ask for missing preprepare messages. You are right, it's not related to this PR, it slipped in by accident.

**Comment by sergefdrv** [comment]
Summary of [changes](https://github.com/filecoin-project/mir/compare/68f352a610535ab3e6ed27cc988dab8611c00240..3b4f3e1cfe9453dd4934cd292e61dea8dd7398c7): squash and rebase

**Review by sergefdrv — APPROVED** [review]
:+1:  Thanks!
